Source/Core/VideoCommon/Assets/CustomTextureData.cpp

```diff
@@ -291,19 +305,35 @@ static bool ParseDDSHeader(File::IOFile& file, DDSLoadInfo* info)
   {
     // Handle DX10 extension header.
     u32 dxt10_format = 0;
+    u32 array_size = 1;
     if (header.ddspf.dwFourCC == MAKEFOURCC('D', 'X', '1', '0'))
     {
       DDS_HEADER_DXT10 dxt10_header;
       if (!file.ReadBytes(&dxt10_header, sizeof(dxt10_header)))
         return false;
 
-      // Can't handle array textures here. Doesn't make sense to use them, anyway.
       if (dxt10_header.resourceDimension != DDS_DIMENSION_TEXTURE2D || dxt10_header.arraySize != 1)
-        return false;
+        array_size = dxt10_header.arraySize;
```

Comment:
The newest revision doesn't have info->array_size = 1 at the top of the quoted section. This creates a code path where it remains 0 when all of the following are true, which seems to be unintended:

- header.ddspf.dwFourCC == MAKEFOURCC('D', 'X', '1', '0')
- dxt10_header.resourceDimension == DDS_DIMENSION_TEXTURE2D
- dxt10_header.arraySize == 1

Since arraySize == 1 is the default, is there any reason not to just initialize it that way in the declaration of DDSLoadInfo?

Reply:
Well, I'd assume that it is possible to have a DDS texture with nothing in it but I've never tried it.  I was going too fast on this change, I will go ahead and look at this again.  Apologies.

Reply:
Ok, looking again, I think that block can just be removed.  That should handle the 'fourcc' case.  Looks like the other cases are also handled.